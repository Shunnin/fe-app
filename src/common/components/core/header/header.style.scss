@import 'style/foundation/config';

.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 40px;
  text-decoration: unset;
  padding: $b-space-2x 0;

  &__title {
    font-size: 32px;
    color: $c-lb7;
  }
}
