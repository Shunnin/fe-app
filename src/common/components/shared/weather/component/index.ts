export { Temperature } from './temperature/temperature.component';
export { WeatherCard } from './weather-card/weather-card.component';
export { WeatherItem } from './weather-item/weather-item.component';
export * from './weather-icon';
