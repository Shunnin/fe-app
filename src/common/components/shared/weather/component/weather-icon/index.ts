export { WeatherIcon } from './weather-icon.component';
export * from './weather-icon.constant';
