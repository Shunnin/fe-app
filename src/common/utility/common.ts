export const getNextSevenDays = (): string[] => {
  const days = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
  const next7Days = [];

  // eslint-disable-next-line no-magic-numbers
  for (let i = 0; i < 7; i++) {
    // eslint-disable-next-line no-magic-numbers
    next7Days.push(days[new Date(Date.now() + (i + 1) * 24 * 60 * 60 * 1000).getDay()]);
  }

  return next7Days;
};
