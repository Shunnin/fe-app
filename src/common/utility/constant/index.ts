export * from './number';
export * from './common';
export * from './url';
