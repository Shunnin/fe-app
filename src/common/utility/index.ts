export * from './route';
export * from './constant';
export * from './promise';
export * from './temperature';
export * from './common';
export * from './url';
