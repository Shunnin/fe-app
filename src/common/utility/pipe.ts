export const pipe =
  (...hofs) =>
  x =>
    hofs.reduceRight((currying, hof) => hof(currying), x);
