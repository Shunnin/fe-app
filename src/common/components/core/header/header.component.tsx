import { FC } from 'react';
import { Link } from 'react-router-dom';

import { URL_HOME } from '../../../utility';

import './header.style';

export const Header: FC = () => {
  return (
    <Link to={URL_HOME} className="header clickable">
      <h1 className="header__title">Welcome, baby</h1>
    </Link>
  );
};
