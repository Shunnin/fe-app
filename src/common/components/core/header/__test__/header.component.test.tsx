import { render, screen } from '@testing-library/react';

import { createMockStore, withRouteProvider } from '../../../../utility/test';

import { Header } from '../header.component';

describe('<Header />', () => {
  it('renders and matches the snapshot', () => {
    const store = createMockStore();
    const { container } = render(withRouteProvider(store)(<Header />));

    const searchBoxEl = container.firstChild;
    const searchBoxPlaceHolder = screen.getByText('Welcome, baby');

    expect(searchBoxEl).toMatchSnapshot();
    expect(searchBoxPlaceHolder).toBeInTheDocument();
  });
});
