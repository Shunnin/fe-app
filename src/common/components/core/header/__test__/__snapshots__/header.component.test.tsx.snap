// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders and matches the snapshot 1`] = `
<a
  class="header clickable"
  href="/"
>
  <h1
    class="header__title"
  >
    Welcome, baby
  </h1>
</a>
`;
